source/Examples/SmartCache/README.md

```diff
@@ -6,108 +6,147 @@ It implements a simple cache with different replacement policies and cache size
+  - [Building](#building)
+    - [Linux](#linux)
+    - [Windows](#windows)
+  - [Executing](#executing)
+    - [Without an optimizer](#without-an-optimizer)
+      - [Linux](#linux-1)
+      - [Windows](#windows-1)
+      - [Example output](#example-output)
+    - [With an optimizer](#with-an-optimizer)
+      - [Example output](#example-output-1)
+  - [Explanation](#explanation)
+  - [See Also](#see-also)
 
-To do that, we run the (C++) `SmartCache` executable to communicate with the (C#) [`Mlos.Agent.Server`](../../Mlos.Agent.Server/#mlos-github-tree-view) over a shared memory channel provided by the [`Mlos.Core`](../../Mlos.Core/#mlos-github-tree-view) library.
-
-The `Mlos.Agent.Server` is essentially a small wrapper around several other communication channels to allow different components to connect for component experimentation convenience.
-
-It provides
+## Building
 
-1. Shared memory communication channels via the [`Mlos.Agent`](../../Mlos.Agent/#mlos-github-tree-view) and [`Mlos.NetCore`](../../Mlos.NetCore/#mlos-github-tree-view) libraries.
+See Also
 
-2. A [`Mlos.Agent.GrpcServer`](../../Mlos.Agent.GrpcClient/#mlos-github-tree-view) GRPC channel to allow driving the experimentation process from a Jupyter notebook.
+- General [build](../../../documentation/02-Build.md) instructions
 
-3. A GRPC client to connect to the (Python) [`mlos.Grpc.OptimizerMicroserviceServer`](../../Mlos.Python/mlos/Grpc/OptimizerMicroserviceServer.py#mlos-github-tree-view) to store and track those experiments.
+> Note: these commands are given relative to the root of the MLOS repo.
 
-TODO: Diagrams
+### Linux
 
-## Building
+```sh
+make -C source/Mlos.Agent.Server
+make -C source/Examples/SmartCache all install
+```
 
-> Note: these commands are given relative to the root of the MLOS repo.
->
-> To move there, you can execute the following within the repository:
->
-> `cd $(git rev-parse --show-toplevel)`
+> The `install` target places the output in the more convenient `target/bin/...` path.
 
-To build and run the necessary components for this example
+### Windows
 
-1. [Build the Docker image](../../../documentation/01-Prerequisites.md#build-the-docker-image) using the [`Dockerfile`](../../../Dockerfile#mlos-github-tree-view) at the root of the repository.
+```cmd
+msbuild /m /r source/Mlos.Agent.Server/Mlos.Agent.Server.csproj
+msbuild /m /r source/Examples/SmartCache/SmartCache.vcxproj
+```
 
-    ```shell
-    docker build --build-arg=UbuntuVersion=20.04 -t mlos/build:ubuntu-20.04 .
-    ```
+## Executing
 
-2. [Run the Docker image](../../../documentation/02-Build.md#create-a-new-container-instance) you just built.
+### Without an optimizer
 
-    ```shell
-    docker run -it -v $PWD:/src/MLOS --name mlos-build mlos/build:ubuntu-20.04
-    ```
+#### Linux
 
-3. Inside the container, [build the compiled software](../../../documentation/02-Build.md#cli-make) with `make`:
+```sh
+dotnet target/bin/Release/Mlos.Agent.Server.dll \
```

Comment:
this actually fits on a single line, right? that would make copy&paste a bit easier.

Reply:
Sure, will fix.